hathor/p2p/netfilter/chain.py

```diff
@@ -30,12 +30,30 @@ def __init__(self, name: str, policy: 'NetfilterTarget'):
         self.rules: List['NetfilterRule'] = []
         self.policy = policy
 
-    def add_rule(self, rule: 'NetfilterRule') -> 'NetfilterChain':
+    def to_json(self) -> Dict[str, Any]:
+        return {
+            'name': self.name,
+            'table': self.table.to_json() if self.table else None,
+            'policy': self.policy.to_json()
+        }
+
+    def add_rule(self, rule: 'NetfilterRule') -> Optional['NetfilterChain']:
```

Comment:
Why return type is `Optional`?

Reply:
It shouldn't, it's a leftover from a previous change that I had to undo.